libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -106,54 +105,15 @@ void UARTDriver::_allocate_buffers(uint16_t rxS, uint16_t txS)
         txS = 32000;
     }
 
-    /*
-      allocate the read buffer
-    */
-    if (rxS != 0 && rxS != _readbuf_size) {
-        _readbuf_size = rxS;
-        if (_readbuf != NULL) {
-            free(_readbuf);
-        }
-        _readbuf = (uint8_t *)malloc(_readbuf_size);
-        _readbuf_head = 0;
-        _readbuf_tail = 0;
-    }
-
-    /*
-      allocate the write buffer
-    */
-    if (txS != 0 && txS != _writebuf_size) {
-        _writebuf_size = txS;
-        if (_writebuf != NULL) {
-            free(_writebuf);
-        }
-        _writebuf = (uint8_t *)malloc(_writebuf_size);
-        _writebuf_head = 0;
-        _writebuf_tail = 0;
-    }
-
-    if (_writebuf_size != 0 && _readbuf_size != 0) {
+    if (_writebuf.set_size(txS) && _readbuf.set_size(rxS)) {
```

Comment:
Again, this is different than before:
- as others, it only changed size if *txS*/*rxS* were different than 0 and current size
- it only set *_initialised* if the buffers had a size different than 0

Now this could be setting the buffers to have size 0 and set *_initialised* to true.